src/cpu.rs

```diff
@@ -72,6 +82,8 @@ impl CpuFlags {
             CpuFlags::AVX512ICL,
             #[cfg(target_feature = "neon")]
             CpuFlags::NEON,
+            #[cfg(any(target_arch = "riscv64", target_arch = "riscv32"))]
```

Comment:
```suggestion
            #[cfg(target_feature = "v")]
```

Reply:
We also need the runtime detection equivalent that can go below the arm ones:

```rust
        #[cfg(any(target_arch = "riscv32", target_arch = "riscv64"))]
        if std::arch::is_riscv_feature_detected!("v") {
            flags |= Self::V;
        }
```

This also requires us changing in `lib.rs`

```rust
#![cfg_attr(target_arch = "arm", feature(stdsimd))]
```

to this

```rust
#![cfg_attr(any(target_arch = "arm", target_arch = "riscv32", target_arch = "riscv64"), feature(stdsimd))]
```

Reply:
It looks like the code changed after you made this suggestion. I used the following:

```
#![cfg_attr(target_arch = "arm", feature(stdarch_arm_feature_detection))]
#![cfg_attr(any(target_arch = "riscv32", target_arch = "riscv64"), feature(stdsimd))]
```

LMK if this seems right, I'm not terribly familiar with Rust feature detection.

Reply:
Yeah that looks right.  They split those features into more fine grained ones.

Reply:
I think we want to keep using this `cfg` (instead of `#[cfg(target_feature = "v")]`) because we want a CPU flags type even if we're on a RISC-V that doesn't have the vector instructions. Only declaring the `V` constant when the feature is detected is also cumbersome since we reference it other places (and need to reference it in more places as the RISC-V assembly gets patched in).

```Rust
#[cfg(any(target_arch = "riscv64", target_arch = "riscv32"))]
bitflags! {
    #[derive(Clone, Copy)]
    pub struct CpuFlags: c_uint {
        const V = 1 << 0;
    }
}
```

Reply:
Yeah, for the type itself, it should be cfg'd on riscv.  But the compile time method is the one where we cfg based on the target features, like we do for the other arches.